app/admin_app.py

```diff
@@ -0,0 +1,83 @@
+from flask import Flask
+from flask import jsonify
+from flask import make_response
+from flask import render_template
+from flask import request
+import json
+
+app = Flask(__name__)
+# set up freezer config
+app.config.from_pyfile('settings.py')
+# read json data
+data = json.loads(open('data.json', 'r').read())
+
+@app.route('/', methods=['GET', 'POST'])
+def index_page():
+    if request.method == 'GET':
+		return render_template('form.html')
+    else:
+        val = request.form.get('button')
+        if val == 'apis':
+            return apis(request)
+        elif val == 'challenges':
+            return challenges(request)
+        else:
+			return render_template('form.html')
```

Comment:
why is this indentation so weird

Reply:
I don't see that in my code... rip. I'll reindent